### ISSUE TITLE
Possible memory leak

### ISSUE THREAD

**Issue by henyxia** [henyxia/slip#2]
It seems that during high rate flooding, the server still work but is not able to close some threads

**Comment by henyxia** [comment]
According to valgrind, some conditional jumps were unsafe

```
==24702== Conditional jump or move depends on uninitialised value(s)
==24702==    at 0x402FF15: __GI___rawmemchr (vg_replace_strmem.c:1241)
==24702==    by 0x40E95B5: _IO_str_init_static_internal (strops.c:44)
==24702==    by 0x40D95A6: __isoc99_vsscanf (isoc99_vsscanf.c:41)
==24702==    by 0x40D951E: __isoc99_sscanf (isoc99_sscanf.c:31)
==24702==    by 0x804944A: interpreter (http.c:54)
==24702==    by 0x8049E84: processHTTPClient (http.c:232)
==24702==    by 0x804AB81: processThread (libthrd.c:27)
==24702==    by 0x4061EFA: start_thread (pthread_create.c:309)
==24702==    by 0x416162D: clone (clone.S:129)
```

**Comment by henyxia** [comment]
According to Valgrind again, some leaks could come from the thread allocation through libthrd and some others from non-closed streams

```
==28005== 12 bytes in 1 blocks are definitely lost in loss record 1 of 2
==28005==    at 0x40291CC: malloc (vg_replace_malloc.c:296)
==28005==    by 0x804A397: newUDPClient (udp.c:125)
==28005==    by 0x804A844: serveurMessages (libcom.c:100)
==28005==    by 0x8049075: startUDPServer (serveur.c:36)
==28005==    by 0x804AB9D: processThread (libthrd.c:27)
==28005==    by 0x4061EFA: start_thread (pthread_create.c:309)
==28005==    by 0x416162D: clone (clone.S:129)
==28005== 
==28005== 352 bytes in 1 blocks are still reachable in loss record 2 of 2
==28005==    at 0x40291CC: malloc (vg_replace_malloc.c:296)
==28005==    by 0x40DB42B: __fopen_internal (iofopen.c:73)
==28005==    by 0x40DB50A: fopen@@GLIBC_2.1 (iofopen.c:103)
==28005==    by 0x8049E6D: processHTTPClient (http.c:230)
==28005==    by 0x804AB9D: processThread (libthrd.c:27)
==28005==    by 0x4061EFA: start_thread (pthread_create.c:309)
==28005==    by 0x416162D: clone (clone.S:129)
```

**Comment by henyxia** [comment]
Stream secured through commit henyxia/slip@60eb1b1b643f94fd6004e18dbad8ec7ad8ce5787

**Comment by henyxia** [comment]
Previous leak was secured in henyxia/slip@309423395ff11daa3559b0f9509fe4360b725e0b
This wasn't a thread allocation issue
There's still an issue with the unconditional jump

**Comment by henyxia** [comment]
According to [this issue](https://github.com/rethinkdb/rethinkdb/issues/2961) it was a false positive
The issue was fixed since Valgrind 1.14
Closing #2
